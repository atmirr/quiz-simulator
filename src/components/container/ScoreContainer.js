import React, {Component} from 'react'
import {connect} from 'react-redux'
import Score from "../ui/Score";

class ScoreContainer extends Component {

    render() {
        const {name, score, history} = this.props
        return (
            <div>
                <Score name={name} score={score} history={history} />
            </div>
        )
    }
}

export const mapStateToProps = state => ({
    name: state.userProfile.name,
    score: state.userProfile.score
})

export default connect(mapStateToProps)(ScoreContainer)