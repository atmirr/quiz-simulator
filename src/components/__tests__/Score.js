import React from 'react'
import {mapStateToProps} from '../container/ScoreContainer'
import Score from '../ui/Score'
import renderer from 'react-test-renderer'

describe("The Score Component", () => {
    describe("The Container Element", () => {
        describe("mapStateToProps", () => {
            it("should map the state to props correctly", () => {
                const sampleUserProfile = {
                    name: 'Amir',
                    score: 20
                }
                const appState = {
                    userProfile: sampleUserProfile
                }
                const componentState = mapStateToProps(appState)
                expect(componentState).toEqual(sampleUserProfile)
            })
        })
    })
    describe("The Display Element", () => {
        it("Should not regress", () => {
            const tree = renderer
                .create(<Score name="Amir" score={20} />)
                .toJSON()
            expect(tree).toMatchSnapshot()
        })
    })
})