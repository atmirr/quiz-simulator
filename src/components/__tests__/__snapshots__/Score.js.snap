// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Score Component The Display Element Should not regress 1`] = `
<div
  className="jumbotron"
>
  <h1>
    You Score
  </h1>
  <p>
    Dear 
    Amir
    ,
  </p>
  <p>
    You got 
    20
     score in this quiz!
  </p>
  <div
    className="row justify-content-md-center"
    style={
      Object {
        "marginTop": 20,
      }
    }
  >
    <button
      className="btn btn-info large"
      onClick={[Function]}
      type="button"
    >
      Try Again
    </button>
  </div>
</div>
`;
